src/main/java/org/elasticsearch/script/expression/ExpressionScript.java

```diff
@@ -109,7 +115,7 @@ public void setNextVar(String name, Object value) {
                 if (value instanceof Number) {
                     specialValue.setValue(((Number)value).doubleValue());
                 } else {
-                    throw new ExpressionScriptExecutionException("Cannot use expression with text variable");
+                    throw new ExpressionScriptExecutionException("Error with " + compiledScript.type().name() + " script [" + compiledScript.name() + "]: Cannot use expression with text variable");
```

Comment:
@jdconrad would it make sense to have a `toString` methdo on CompiledScript that includes name and type?